src/test/scala/com/lucidworks/spark/MovieLensTestSuite.scala

```diff
@@ -5,6 +5,40 @@ import org.apache.spark.sql.types.DoubleType
 
 class MovieLensTestSuite extends MovielensBuilder {
 
+  test("multiple nested where clauses with NOT and AND") {
+    val sql =
+      s"""
+         | select movie_id, title from ${moviesColName} m where
+         |     ((m.genre IN ('comedy') and (m.title != 'Here Comes Cookie (1935)')))
+         |     OR
+         |     (m.genre IN ('action') and m.title = 'Operation Dumbo Drop (1995)')
+        """.stripMargin
+    val results = sparkSession.sql(sql).collect()
+    assert(results.length > 1)
```

Comment:
[0] Is it possible/useful to make more specific assertions about the results that come back?

As-is, this test ensures that spark-solr doesn't blow up in this case anymore.  Which is an improvement.  But since we're not really asserting anything here, it falls short of actually verifying that we handle the NOT correctly.